Data Structure 2 Assignment/Counseling Process/src/com/metacube/controller/Controller.java

```diff
@@ -0,0 +1,79 @@
+package com.metacube.controller;
+
+import com.metacube.facade.CounselingFacade;
+import com.metacube.main.View;
+
+/**
+ * The Class Controller controls the counseling facade.
+ */
+public class Controller {
+	private CounselingFacade counselingFacade;
+	private static Controller controller;
+
+	/**
+	 * Instantiates a new controller.
+	 */
+	private Controller() {
+		counselingFacade = CounselingFacade.getInstance();
+	}
+
+	/**
+	 * Gets the single instance of Controller.
+	 *
+	 * @return single instance of Controller
+	 */
+	public static Controller getInstance() {
```

Comment:
Generally, Controller shouldn't use singleton pattern because Controller is directly responsible for maintaining user-specific data.